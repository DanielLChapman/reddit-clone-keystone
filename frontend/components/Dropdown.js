import React, { useState } from 'react';
import PropTypes from 'prop-types'

function Dropdown(props) {
    /*
        let options = ['Hot', 'Best', 'New', 'Top', 'Rising'];
        const [dropValue, setDropValue] = useState('Hot');
        <Dropdown returnFunc={setDropValue} selected={dropValue} options={options} />    
    */
    return (
        <select value={props.selected} onChange={(e) => {
            props.returnFunc(e.target.value);
        }}>
            {
                props.options && props.options.map((x, i) => (
                    <option key={i} value={x}>{x}</option>
                ))
            }
        </select>
    );
}

Dropdown.propTypes = {
    returnFunc: PropTypes.func.isRequired,
    options: PropTypes.array.isRequired,
    selected: PropTypes.string.isRequired,
};
  

export default Dropdown;