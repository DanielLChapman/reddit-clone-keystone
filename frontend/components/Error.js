import React from 'react';

function Error(props) {
    return (
        <span>
            {props.error}
        </span>
    );
}

export default Error;