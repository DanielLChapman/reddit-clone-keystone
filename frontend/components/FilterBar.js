import React, { Component } from 'react';
import { FaHotjar, FaBahai, FaGopuram, FaChartLine} from 'react-icons/fa';



class FilterBar extends Component {
    constructor(props) {
        super(props);
        this.state = {
            li1: true,
            li2: false,
            li3: false,
            li4: false
        }
    }

    toggleClass = (num) => {
        let state = {
            li1: false,
            li2: false,
            li3: false,
            li4: false
        };
        state['li'+num] = true;
        this.setState({...state});
    }

    render() {
        let style = {
            display: "none"
        };
        if (this.state.li3) {
            style= {
                display: "inline-block"
            };
        }
        return (
            <section className="filter-bar"> 
                <h5>Popular Posts</h5>
                <ul>
                    <li className={this.state.li1 ? 'active' : null} onClick={() => this.toggleClass(1)}><FaHotjar /> Hot</li>
                    <li className={this.state.li2 ? 'active' : null} onClick={() => this.toggleClass(2)}><FaBahai /> New</li>
                    <li className={this.state.li3 ? 'active' : null} onClick={() => this.toggleClass(3)}><FaGopuram /> Top</li>
                    <li className="time-selector-for-filter-bar" style={style}>Today</li>
                    <li className={this.state.li4 ? 'active' : null} onClick={() => this.toggleClass(4)}><FaChartLine /> Rising</li>
                </ul>
            </section>
        );
    }
}

export default FilterBar;