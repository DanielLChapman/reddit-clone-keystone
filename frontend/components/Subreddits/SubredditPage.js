import { useQuery } from '@apollo/client';
import gql from 'graphql-tag';
import React from 'react';
import SubredditContent from './SubredditContent';

const GET_SUBREDDIT_INFO = gql`
    query GET_SUBREDDIT_INFO($slug: String!) {
        allSubreddits(where: {
            slug: $slug
        }) {
            name
            title
            sidebar
            description
            status
            owner {
                name
            }
            moderators {
                name
            }
        }
    }
`;

function SubredditPage(props) {

    const {data, error, loading} = useQuery(GET_SUBREDDIT_INFO, {
        variables: {
            slug: props.slug
        }
    });
    if (loading) return <span>Loading...</span>
    if(error) return <span>Nothing To See Here</span>

    console.log(data);
    return (
        <div className="subreddit-content">
            <div className="subreddit-left">
                {/* HArd coding news id, will grab from gql later */}
                <SubredditContent slug={props.slug} type={props.type} selftext={props.selftext} id='60944706438ea508cdf752c9' /> 

            </div>
            <div className="subreddit-right">
                blah
            </div>
        </div>
    );
}

export default SubredditPage;