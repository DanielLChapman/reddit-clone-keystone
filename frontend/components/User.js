import { gql, useQuery } from '@apollo/client';

export const CURRENT_USER_QUERY = gql`
  query {
    authenticatedItem {
      ... on User {
        id
        username
        email
        name
        posts {
          votes {
            vflag
          }
        }
        comments {
          votes {
            vflag
          }
        }
        moderating {
          id
          slug
          name
        }
        owner {
          id
          slug
          name
        }
        subreddits {
          id
          slug
          name
        }
        postvotes {
          post {
            title
            id
          }
          id
          vflag
        }
        commentvotes {
          comment {
            id
          }
          id
          vflag
        }
    }
  }
  }
`;

export function useUser() {

  const { data } = useQuery(CURRENT_USER_QUERY);
  let test = {...data};

  
  if (test.authenticatedItem) {
    let x = Object.assign({}, test.authenticatedItem);
    //count up post votes
    let postVotes = 0;
    let commentVotes = 0;
    
    x.posts.forEach((q) => {
      q.votes.forEach((y) => {
        y.vflag === 'Upvote' ? postVotes += 1 : postVotes -= 1;
      })
    });
    //count up comment votes
    x.comments.forEach((q) => {
      q.votes.forEach((y) => {
        y.vflag === 'Upvote' ? commentVotes += 1 : commentVotes -= 1;
      })
    });

    x.postVotes = postVotes;
    x.commentVotes = commentVotes;

    return x;
  } else {
    return data?.authenticatedItem;
  }

  
}
