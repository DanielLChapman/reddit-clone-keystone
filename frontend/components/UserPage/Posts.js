import React from 'react';

function Posts(props) {
    return (
        <div>
            posts
        </div>
    );
}


export default Posts;