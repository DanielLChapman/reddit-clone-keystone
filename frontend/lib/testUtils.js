import casual from 'casual';

  
// seed it so we get consistent results
casual.seed(777);

const fakeUser = (overrides) => (
    {
    __typename: 'User',
    id: '4234',
    name: casual.name,
    username: casual.name,
    owner: [],
    email: casual.email,
    permissions: ['ADMIN'],
    moderating: [],
    subreddits: [],
    posts: [],
    postvotes: [],
    firstSignin: false,
    comments: [],
    commentvotes:[],
    ...overrides,
});

const fakeSubreddit = (overrides) => (
    {
        __typename: 'Subreddit',
        id: casual.uuid,
        name: casual.string,
        title: casual.string,
        slug: casual.string,
        sidebar: casual.string,
        rules: casual.string,
        description: casual.string,
        status: 'PUBLIC',
        ownder: fakeUser(),
        moderators: [],
        subscriber: [],
        posts: [],
        createdAt: Date.now(),
    }
);

const fakeLinkPost = (overrides) => (
    {
    __typename: 'Post',
    title: casual.sentence,
    user: fakeUser(),
    subreddit: fakeSubreddit(),
    createdAt: Date.now(),
    votes: [],
    post_slug: casual.string,
    link: 'www.google.com',
    type: 'Link',
    comments: [],
    removed: 'False',
    id: casual.uuid,


});

const fakeMediaPost = (overrides) => (
    {
    __typename: 'Post',
    title: casual.sentence,
    user: fakeUser(),
    subreddit: fakeSubreddit(),
    createdAt: Date.now(),
    votes: [],
    post_slug: casual.string,
    link: 'https://cdn.fileinfo.com/img/ss/sm/png_79.jpg',
    type: 'Link',
    comments: [],
    id: casual.uuid,
    removed: 'False'

});

const fakeTextPost = (overrides) => (
    {
    __typename: 'Post',
    title: casual.sentence,
    content: casual.sentence,
    user: fakeUser(),
    subreddit: fakeSubreddit(),
    createdAt: Date.now(),
    votes: [],
    post_slug: casual.string,
    link: '',
    type: 'Text',
    comments: [],
    id: casual.uuid,
    removed: 'False'

});

const fakeComment = (overrides) => (
    {
    __typename: 'Comment',
    content: casual.sentence,
    user: fakeUser(),
    post: fakeTextPost(),
    createdAt: Date.now(),
    votes: [],
    parent: null,
    id: casual.uuid,
    children: [],

});


// Fake LocalStorage
class LocalStorageMock {
    constructor() {
      this.store = {};
    }
  
    clear() {
      this.store = {};
    }
  
    getItem(key) {
      return this.store[key] || null;
    }
  
    setItem(key, value) {
      this.store[key] = value.toString();
    }
  
    removeItem(key) {
      delete this.store[key];
    }
  }



export {

    LocalStorageMock,
    fakeLinkPost,
    fakeComment,
    fakeMediaPost,
    fakeSubreddit,
    fakeTextPost,
    fakeUser,

  };