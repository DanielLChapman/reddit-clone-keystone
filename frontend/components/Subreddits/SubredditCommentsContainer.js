import React from 'react';
import PropTypes from 'prop-types';
import gql from 'graphql-tag';
import { FaLongArrowAltUp, FaLongArrowAltDown } from 'react-icons/fa';
import { useMutation, useQuery } from '@apollo/client';
import { CREATE_VOTE, DELETE_VOTE, UPDATE_VOTE } from '../SmallPosts/PostMain';
import { CURRENT_USER_QUERY, useUser } from '../User';
import CommentsPage from './SubredditComments/CommentsPage';
import SubredditTopBar from './SubredditComments/CommentsPageTopBar';

const GET_POST_INFO = gql`
    query GET_POST_INFO($id: ID!) {
        Post( where: { id: $id } )
    {
        id
        content
        title
        user {
            id
            name
            username
        }
        createdAt
        votes {
            vflag
        }
        post_slug
        link
        removed
        subreddit {
            slug
        }
        type
        comments {
            id
            parent {
                id
            }
            content
            createdAt
            user {
                id
                username
            }
            votes {
                vflag
            }
        }

    }
    }
`;

const GET_SPECIFIC_COMMENTS = gql`
    query GET_SPECIFIC_COMMENTS($id: ID!) {
        Comment (
            where: {
                id: $id
            }
        ) {
            id
            parent {
                id
            }
            content
            createdAt
            user {
                id
                username
            }
            votes {
                vflag
            }
            children {
                id
                parent {
                    id
                }
                content
                createdAt
                user {
                    id
                    username
                }
                votes {
                    vflag
                }
                children {
                    id
                    parent {
                        id
                    }
                    content
                    createdAt
                    user {
                        id
                        username
                    }
                    votes {
                        vflag
                    }
                    children {
                        id
                        parent {
                            id
                        }
                        content
                        createdAt
                        user {
                            id
                            username
                        }
                        votes {
                            vflag
                        }
                        children {
                            id
                            parent {
                                id
                            }
                            content
                            createdAt
                            user {
                                id
                                username
                            }
                            votes {
                                vflag
                            }
                        }
                    }
                }
            }
        } 
    }
`;

function SubredditCommentsContainer(props) {
    const user = useUser();

    const {data, error, loading} = useQuery(GET_POST_INFO, {
        variables: {
            id: props.postid,
        }
    });

    

    if (error) return <span>Error...</span>
    if (loading) return <span>Loading...</span>

    let post = {...data?.Post};
    let style = {
        display: 'block'
    }
    let showLinkToGoBackToAllComments = false;

    if (props.commentArray) {
        post.comments = props.commentArray;
        showLinkToGoBackToAllComments = true;
        style = {
            display: 'none'
        }
    }

    return (
        <div className="comment-page">
            <SubredditTopBar user={user} post={post} subreddit={props.subreddit}/>
            <section className="comment-container" >
                
                <CommentsPage showLinkToGoBackToAllComments={showLinkToGoBackToAllComments} propstyle={style} ownership={props.ownership} user={user} post={post} subreddit={props.subreddit} />
            </section>
        </div>
    );
}

// SubredditCommentsContainer.propTypes = {
//     subreddit: PropTypes.object.isRequired,
//     postid: PropTypes.string.isRequired,

// }

export default SubredditCommentsContainer;