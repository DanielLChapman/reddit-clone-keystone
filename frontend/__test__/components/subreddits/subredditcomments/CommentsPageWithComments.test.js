import { act, render, screen, waitFor } from "@testing-library/react";
import { MockedProvider } from "@apollo/react-testing";
import wait from "waait";
import { fakeComment, fakeTextPost, fakeUser } from "../../../../lib/testUtils";
import { GET_SUBREDDIT_INFO } from "../../../../components/Subreddits/SubredditPage";
import Posts from "../../../../pages/r/[slug]/comments/[id]/[post_slug]";
import { CURRENT_USER_QUERY } from "../../../../components/User";
import { GET_POST_INFO } from "../../../../components/Subreddits/SubredditCommentsContainer";
import userEvent from "@testing-library/user-event";
import gql from "graphql-tag";

const myMock = jest.fn();

let fakePostObj = {
    ...fakeTextPost(),
    comments: [
        {
            ...fakeComment(),
        },
    ],
};
let subreddit = fakePostObj.subreddit;
const fakeUserObj = {
    ...fakeUser(),
    owner: [
        {
            id: subreddit.id,
            name: subreddit.name,
            slug: subreddit.name.toLowerCase(),
        },
    ],
};
let fakeCommentObj = { ...fakeComment(), post: { ...fakePostObj } };

//Error with importing these two*

const CREATE_VOTE = gql`
    mutation CREATE_VOTE($post_id: ID!, $vflag: String!) {
        createCommentVote(
            data: { comment: { connect: { id: $post_id } }, vflag: $vflag }
        ) {
            id
            vflag
        }
    }
`;
const SUBMIT_POST_WITHOUT_PARENT_MUTATION = gql`
    mutation SUBMIT_POST_WITHOUT_PARENT_MUTATION(
        $post_id: ID!
        $content: String!
    ) {
        createComment(
            data: { content: $content, post: { connect: { id: $post_id } } }
        ) {
            id
            content
            createdAt
            parent {
                id
            }
            user {
                id
                username
            }
            votes {
                vflag
            }
        }
    }
`;

const signedOutMocks = [

    {
        request: {
            query: GET_SUBREDDIT_INFO,
            variables: {
                slug: fakePostObj.subreddit.name.toLowerCase(),
            },
        },
        result: () => {
            return {
                data: {
                    allSubreddits: [
                        {
                            id: subreddit.id,
                            name: subreddit.name,
                            title: subreddit.title,
                            slug: subreddit.name.toLowerCase(),
                            sidebar: subreddit.sidebar,
                            description: subreddit.description,
                            status: "Public",
                            owner: {
                                username: subreddit.owner.username,
                            },
                            moderators: [],
                            createdAt: subreddit.createdAt,
                        },
                    ],
                },
            };
        },
    },

    {
        request: {
            query: GET_POST_INFO,
            variables: {
                id: fakePostObj.id,
            },
        },
        result: {
            data: {
                Post: {
                    id: fakePostObj.id,
                    content: fakePostObj.content,
                    title: fakePostObj.title,
                    user: {
                        id: fakePostObj.user.id,
                        name: fakePostObj.user.name,
                        username: fakePostObj.user.username,
                    },
                    createdAt: fakePostObj.createdAt,
                    votes: [],
                    post_slug: fakePostObj.post_slug,
                    link: fakePostObj.link,
                    removed: fakePostObj.removed,
                    subreddit: {
                        slug: fakePostObj.subreddit.slug,
                    },
                    type: fakePostObj.type,
                    comments: fakePostObj.comments,
                },
            },
        },
    },
    {
        request: {
            query: SUBMIT_POST_WITHOUT_PARENT_MUTATION,
            variables: {
                post_id: fakePostObj.id,
                content: fakeCommentObj.content,
            },
        },
        result: {
            data: {
                createComment: {
                    id: fakeCommentObj.id,
                    content: fakeCommentObj.content,
                    createdAt: fakeCommentObj.createdAt,
                    parent: { id: 1 },
                    user: {
                        id: fakeUserObj.id,
                        username: fakeUserObj.username,
                    },
                    votes: {
                        vflag: [],
                    },
                },
            },
        },
    },
    {
        request: {
            query: CREATE_VOTE,
            variables: {
                post_id: fakePostObj.id,
                vflag: "Upvote",
            },
        },
        result: {
            data: {
                createCommentVote: {
                    id: "123456789createCommentVote",
                    vflag: "Upvote",
                },
            },
        },
    },
];

const signedInMocks = [
    {
        request: { query: CURRENT_USER_QUERY },
        result: {
            data: {
                authenticatedItem: {
                    ...fakeUserObj,
                },
            },
        },
    },
    {
        request: {
            query: GET_SUBREDDIT_INFO,
            variables: {
                slug: fakePostObj.subreddit.name.toLowerCase(),
            },
        },
        result: () => {
            return {
                data: {
                    allSubreddits: [
                        {
                            id: subreddit.id,
                            name: subreddit.name,
                            title: subreddit.title,
                            slug: subreddit.name.toLowerCase(),
                            sidebar: subreddit.sidebar,
                            description: subreddit.description,
                            status: "Public",
                            owner: {
                                username: subreddit.owner.username,
                            },
                            moderators: [],
                            createdAt: subreddit.createdAt,
                        },
                    ],
                },
            };
        },
    },

    {
        request: {
            query: GET_POST_INFO,
            variables: {
                id: fakePostObj.id,
            },
        },
        result: {
            data: {
                Post: {
                    id: fakePostObj.id,
                    content: fakePostObj.content,
                    title: fakePostObj.title,
                    user: {
                        id: fakePostObj.user.id,
                        name: fakePostObj.user.name,
                        username: fakePostObj.user.username,
                    },
                    createdAt: fakePostObj.createdAt,
                    votes: [],
                    post_slug: fakePostObj.post_slug,
                    link: fakePostObj.link,
                    removed: fakePostObj.removed,
                    subreddit: {
                        slug: fakePostObj.subreddit.slug,
                    },
                    type: fakePostObj.type,
                    comments: fakePostObj.comments,
                },
            },
        },
    },
    {
        request: {
            query: SUBMIT_POST_WITHOUT_PARENT_MUTATION,
            variables: {
                post_id: fakePostObj.id,
                content: fakeCommentObj.content,
            },
        },
        result: {
            data: {
                createComment: {
                    id: fakeCommentObj.id,
                    content: fakeCommentObj.content,
                    createdAt: fakeCommentObj.createdAt,
                    parent: { id: 1 },
                    user: {
                        id: fakeUserObj.id,
                        username: fakeUserObj.username,
                    },
                    votes: {
                        vflag: [],
                    },
                },
            },
        },
    },
    {
        request: {
            query: CREATE_VOTE,
            variables: {
                post_id: fakePostObj.id,
                vflag: "Upvote",
            },
        },
        result: {
            data: {
                createCommentVote: {
                    id: "123456789createCommentVote",
                    vflag: "Upvote",
                },
            },
        },
    },
];

let query = {
    slug: fakePostObj.subreddit.name.toLowerCase(),
    id: fakePostObj.id,
    post_slug: fakePostObj.post_slug,
};


describe("It accurately displays the comment page", () => {
    it("renders a post page with a comment on it", async () => {
        const { container, debug } = render(
            <MockedProvider mocks={signedInMocks}>
                <Posts query={query} />
            </MockedProvider>
        );



        await screen.findByText("Posted 0 minutes ago");
        expect(container).toMatchSnapshot();
        expect(container).toHaveTextContent(fakePostObj.comments[0].content);
    });

    it('renders a comment box under a comment when you are signed in', async () => {
        const { container, debug } = render(
            <MockedProvider mocks={signedInMocks}>
                <Posts query={query} />
            </MockedProvider>

        );
        await screen.findByText("Posted 0 minutes ago");
        let a = await screen.findAllByPlaceholderText('What are your thoughts?');
        expect(a.length).toBe(1);

        await userEvent.click(screen.getByText('Reply'));
        await waitFor(() => wait(0));

        let b = await screen.findAllByPlaceholderText('What are your thoughts?');
        expect(b.length).toBe(2);
    });

    it('does not render reply when not signed in', async () => {
        const { container, debug } = render(
            <MockedProvider mocks={signedOutMocks}>
                <Posts query={query} />
            </MockedProvider>

        );
        await screen.findByText("Posted 0 minutes ago");

        expect(screen.queryByText('Reply')).not.toBeInTheDocument()

        expect(screen.queryByText('Permalink')).toBeInTheDocument();

    })
});
