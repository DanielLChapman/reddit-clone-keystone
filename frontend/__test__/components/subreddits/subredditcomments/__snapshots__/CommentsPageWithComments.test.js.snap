// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It accurately displays the comment page renders a post page with a comment on it 1`] = `
<div>
  <div>
    <div
      class="comment-page"
    >
      <div
        class="comment-top-bar"
      >
        <section
          class="comment-arrows"
        >
          <svg
            class="upvote-arrow "
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 256 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M88 166.059V468c0 6.627 5.373 12 12 12h56c6.627 0 12-5.373 12-12V166.059h46.059c21.382 0 32.09-25.851 16.971-40.971l-86.059-86.059c-9.373-9.373-24.569-9.373-33.941 0l-86.059 86.059c-15.119 15.119-4.411 40.971 16.971 40.971H88z"
            />
          </svg>
          <span>
            0
          </span>
          <svg
            class="downvote-arrow "
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 256 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M168 345.941V44c0-6.627-5.373-12-12-12h-56c-6.627 0-12 5.373-12 12v301.941H41.941c-21.382 0-32.09 25.851-16.971 40.971l86.059 86.059c9.373 9.373 24.569 9.373 33.941 0l86.059-86.059c15.119-15.119 4.411-40.971-16.971-40.971H168z"
            />
          </svg>
        </section>
        <section
          class="post-title"
        >
          Facilis corporis sed
          ...
        </section>
        <section
          class="post-close"
        >
          <a
            href="/r/ipsatemporaautsintaperiamoccaecatiillum"
          >
            × Close
          </a>
        </section>
      </div>
      <section
        class="comment-container"
      >
        <section
          class="comment-page-container"
        >
          <section
            class="comment-page-left"
          >
            <section
              class="reddit-post"
            >
              <section
                class="reddit-post-left"
              >
                <svg
                  class="upvote-arrow "
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 256 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M88 166.059V468c0 6.627 5.373 12 12 12h56c6.627 0 12-5.373 12-12V166.059h46.059c21.382 0 32.09-25.851 16.971-40.971l-86.059-86.059c-9.373-9.373-24.569-9.373-33.941 0l-86.059 86.059c-15.119 15.119-4.411 40.971 16.971 40.971H88z"
                  />
                </svg>
                <div
                  aria-disabled="false"
                  class="upvote-arrow-box"
                />
                <br />
                <span>
                  0
                </span>
                <br />
                <svg
                  class="upvote-arrow "
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 256 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M168 345.941V44c0-6.627-5.373-12-12-12h-56c-6.627 0-12 5.373-12 12v301.941H41.941c-21.382 0-32.09 25.851-16.971 40.971l86.059 86.059c9.373 9.373 24.569 9.373 33.941 0l86.059-86.059c15.119-15.119 4.411-40.971-16.971-40.971H168z"
                  />
                </svg>
                <div
                  aria-disabled="false"
                  class="downvote-arrow-box"
                />
              </section>
              <section
                class="reddit-post-right"
              >
                <section
                  class="reddit-post-right-top"
                >
                  <span>
                    Posted by 
                    <a
                      class="subreddit-link"
                      href="/user/Mrs. Gideon Block"
                    >
                      u/
                      Mrs. Gideon Block
                    </a>
                  </span>
                   
                  <span>
                    Posted 
                    0 minutes ago
                  </span>
                </section>
                <section
                  class="reddit-post-right-bottom"
                >
                  <h6>
                    Facilis corporis sed veritatis.
                  </h6>
                  <div
                    style="display: block;"
                  >
                    <p>
                      Praesentium magnam fugiat error molestiae laborum est.
                    </p>
                  </div>
                </section>
                <section
                  class="reddit-post-right-bottom-footer"
                >
                  <a
                    class="subreddit-link"
                    href="/r/ipsatemporaautsintaperiamoccaecatiillum/comments/7760116b-dd26-46da-a05a-a010e05cef40/explicabo dolores ipsam laboriosam non itaque voluptas"
                  >
                    1
                     Comments
                  </a>
                  
                </section>
              </section>
            </section>
            <section
              class="comments-container"
            >
              <span>
                Comment as 
                Ms. Josh Stamm
              </span>
              <section
                class="comments-submission-container"
              >
                <div>
                  <form>
                    <fieldset
                      data-testid="comments-box"
                    >
                      <label
                        for="content"
                      >
                        <textarea
                          id="content"
                          name="content"
                          placeholder="What are your thoughts?"
                        />
                      </label>
                      <br />
                      <button
                        class="comment-box-submit"
                        type="submit"
                      >
                        Submit
                      </button>
                    </fieldset>
                  </form>
                </div>
              </section>
            </section>
            <div
              class="comment-sorting"
            >
              <span>
                Sort By 
                <select>
                  <option
                    value="Hot"
                  >
                    Hot
                  </option>
                  <option
                    value="Best"
                  >
                    Best
                  </option>
                  <option
                    value="New"
                  >
                    New
                  </option>
                  <option
                    value="Top"
                  >
                    Top
                  </option>
                  <option
                    value="Rising"
                  >
                    Rising
                  </option>
                </select>
                    
              </span>
            </div>
            <div
              class="comment-tree-beginning"
            >
              <div
                class="comments-threadline-container"
              >
                <div
                  class="threadline"
                />
                <div
                  class="comments-indent comments-indent-1"
                >
                  <section
                    class="comment-top"
                  >
                    <span>
                      Posted by 
                      <a
                        class="subreddit-link"
                        href="/user/Ms. Josh Stamm"
                      >
                        u/
                        Ms. Josh Stamm
                      </a>
                    </span>
                     
                     ·  
                    <span>
                      0 minutes ago
                    </span>
                     
                     
                  </section>
                  <section
                    class="comment-content"
                  >
                    <p>
                      Qui aut nobis quo consequatur.
                    </p>
                  </section>
                  <section
                    class="comment-bottom"
                  >
                    <section
                      class="comment-arrows"
                    >
                      <svg
                        class="upvote-arrow "
                        fill="currentColor"
                        height="1em"
                        stroke="currentColor"
                        stroke-width="0"
                        viewBox="0 0 256 512"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M88 166.059V468c0 6.627 5.373 12 12 12h56c6.627 0 12-5.373 12-12V166.059h46.059c21.382 0 32.09-25.851 16.971-40.971l-86.059-86.059c-9.373-9.373-24.569-9.373-33.941 0l-86.059 86.059c-15.119 15.119-4.411 40.971 16.971 40.971H88z"
                        />
                      </svg>
                      <span
                        class="total"
                      >
                        0
                      </span>
                      <svg
                        class="downvote-arrow "
                        fill="currentColor"
                        height="1em"
                        stroke="currentColor"
                        stroke-width="0"
                        viewBox="0 0 256 512"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M168 345.941V44c0-6.627-5.373-12-12-12h-56c-6.627 0-12 5.373-12 12v301.941H41.941c-21.382 0-32.09 25.851-16.971 40.971l86.059 86.059c9.373 9.373 24.569 9.373 33.941 0l86.059-86.059c15.119-15.119 4.411-40.971-16.971-40.971H168z"
                        />
                      </svg>
                    </section>
                    <span>
                      Reply
                    </span>
                    <span>
                      Edit
                    </span>
                    <span>
                      Delete
                    </span>
                    <span>
                      <a
                        href="/r/illum saepe ut assumenda sint quia sed/comments/7760116b-dd26-46da-a05a-a010e05cef40/explicabo dolores ipsam laboriosam non itaque voluptas/960c06ee-001a-4d55-a0c9-954ae468f9ab"
                      >
                        Permalink
                      </a>
                    </span>
                  </section>
                  <section
                    class="comment-children"
                  />
                </div>
              </div>
            </div>
          </section>
          <section
            class="comment-page-right"
          />
        </section>
      </section>
    </div>
  </div>
</div>
`;
